fw/sock.c

```diff
@@ -559,7 +559,8 @@ ss_send(struct sock *sk, struct sk_buff **skb_head, int flags)
 	 */
 	if (unlikely(!ss_sock_active(sk))) {
 		T_DBG2("Attempt to send on inactive socket %p\n", sk);
-		ss_skb_queue_purge(skb_head);
+		if (!(flags & SS_F_KEEP_SKB))
+			ss_skb_queue_purge(skb_head);
```

Comment:
For responses we don't set set `SS_F_KEEP_SKB`, however the issue might affect responses as well. 

[Here](https://github.com/tempesta-tech/tempesta/blob/3bacd900f7dfa1905964aad150f391aedbc8e538/fw/http.c#L4783) we free unsent responses, however if `__tfw_http_resp_fwd() -> tfw_cli_conn_send()` failed and skbs were purged we trying to double free skb_head. Isn't?

Reply:
Oh I see we do check before free